### PR TITLE
Support Workspace symbols

### PR DESCRIPTION
**Pull request by yorkxin**
Support global symbols searching

## Bugs to fix in the next version

- [ ] Git ignored files are included in SymbolIndex
- [x] Performance issue - High CPU load when indexing lots of files (1 minute load average over 300)

## Further improvements

- [ ] Searching for symbols should include JavaScript files

## Checklist

- [x] `client/CHANGELOG.md`
- [ ] Set version to `0.2.0`

### CODE REVIEW

**Comment by carloshpds** [comment]
Any update about this PR? I would like to make some a PR too and i guess that should be important to have this code first.

**Comment by yorkxin** [comment]
@carloshpds Hi, I found that the current beta does not work on VSCode anymore. I need to fix that before I can merge this.

However, if your improvements are related to this pull request, you can send a pull request to this branch `workspace-symbols`. Or, if that is unrelated, feel free to send it to `master`.

Really appreciate your help!

File: server/package.json [hunk 1]
```diff
@@ -12,20 +12,23 @@
     "url": "https://github.com/chitsaou/vscode-coffeescript-support"
   },
   "dependencies": {
-    "coffeescript": "^2.1.0",
-    "vscode-languageserver": "^3.4.2"
+    "coffeescript-symbols": "file:../coffeescript-symbols",
+    "coffeescript-symbol-index": "file:../coffeescript-symbol-index",
```
Comment (yorkxin) [inline server/package.json]: Problem: these packages won't be `npm install`'d during `npm run compile`.

Comment (yorkxin) [reply server/package.json]: Even if unit test passes, the built vsix won't run due to lack of these packages.

Publishing them to NPM could solve the problem, but that's overkill.

**Comment by yorkxin** [comment]
Beta 7 is available https://github.com/chitsaou/vscode-coffeescript-support/releases/tag/v2.0.0-beta.7

**Comment by yorkxin** [comment]
High CPU load problem still exists. Last time it crashed my MBP 2017.

File: server/src/server.ts [hunk 2]
```diff
@@ -4,46 +4,71 @@
...
+  const diagnostics = documentParser.validateSource(change.document.getText());
   connection.sendDiagnostics({ uri: change.document.uri, diagnostics });
 });
 
+connection.onDidChangeWatchedFiles((params) => {
+  params.changes.forEach((change) => {
+    if (change.type === FileChangeType.Deleted) {
+      indexService.removeFiles([change.uri]);
+    } else if (change.type === FileChangeType.Changed || change.type === FileChangeType.Created) {
+      indexService.indexFilesInBackground([change.uri]);
```
Comment (yorkxin) [inline server/src/server.ts]: I think this is causing fork bomb when git checkout between branches that has lots changes 💣

Comment (yorkxin) [reply server/src/server.ts]: Solved by deamonizing the indexer process.

**Pull request merged**